### ISSUE TITLE
Unable to parse lutris error

### ISSUE THREAD

**Issue by Turpster** [RobLoach/lutris-kodi-addon#47]
When trying to launch the addon, it throws this.

```
...
2020-08-31 02:17:56.419 T:139670712051456   ERROR: GetDirectory - Error getting include
2020-08-31 02:17:56.499 T:139670676420352   ERROR: Previous line repeats 27 times.
2020-08-31 02:17:56.499 T:139670676420352   ERROR: GetDirectory - Error getting Submenu
2020-08-31 02:17:56.499 T:139670676420352   ERROR: GetDirectory - Error getting include
2020-08-31 02:17:57.762 T:139668859094784   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: Extra data: line 38 column 1 - line 39 column 1 (char 917 - 969)
                                            Traceback (most recent call last):
                                              File "/home/turpster/.kodi/addons/script.lutris/default.py", line 294, in <module>
                                                router(sys.argv[2][1:])
                                              File "/home/turpster/.kodi/addons/script.lutris/default.py", line 288, in router
                                                list_games()
                                              File "/home/turpster/.kodi/addons/script.lutris/default.py", line 166, in list_games
                                                games = get_games()
                                              File "/home/turpster/.kodi/addons/script.lutris/default.py", line 149, in get_games
                                                response = json.loads(result)
                                              File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
                                                return _default_decoder.decode(s)
                                              File "/usr/lib/python2.7/json/decoder.py", line 367, in decode
                                                raise ValueError(errmsg("Extra data", s, end, len(s)))
                                            ValueError: Extra data: line 38 column 1 - line 39 column 1 (char 917 - 969)
                                            -->End of Python script error report<--
2020-08-31 02:17:57.804 T:139670712051456   ERROR: GetDirectory - Error getting plugin://script.lutris/
2020-08-31 02:18:55.921 T:139669937297152  NOTICE: script.tv.show.next.aired: ### no prior data found
2020-08-31 02:18:57.921 T:139669937297152  NOTICE: script.tv.show.next.aired: ### starting data update
2020-08-31 02:18:57.921 T:139669937297152  NOTICE: script.tv.show.next.aired: ### grabbing a new country mapping list
2020-08-31 02:23:57.683 T:139669945689856 WARNING: Repository has MD5 hashes enabled - this hash function is broken and will only guard against unintentional data corruption
...
```

The error shows us that there is extra data of `969-917=52` characters.
```
ValueError: Extra data: line 38 column 1 - line 39 column 1 (char 917 - 969)
```

And when I execute `lutris --list-games --installed --json`

```
Unable to load locale dir, translations won't work.
2020-08-31 11:06:15,546: MAME XML generation launched in the background, not returning anything this time
2020-08-31 11:06:15,546: Getting full game list from MAME...
2020-08-31 11:06:15,547: MAME isn't installed, can't retrieve systems list.
2020-08-31 11:06:15,837: Running Lutris 0.5.7.1
2020-08-31 11:06:15,837: Using NVIDIA drivers 450.57 for x86_64
2020-08-31 11:06:15,837: GPU: GeForce GTX 970
2020-08-31 11:06:15,837: GPU: 10DE:13C2 10DE:13C2 using nvidia drivers
2020-08-31 11:06:15,880: Vulkan is supported
[
  {
    "id": 2,
    "slug": "bulletstorm-full-clip-edition",
    "name": "Bulletstorm Full Clip Edition",
    "runner": "winesteam",
    "directory": "/run/media/turpster/Files/Games/steamapps/common/Bulletstorm Full Clip Edition"
  },
  {
    "id": 5,
    "slug": "godfather-2",
    "name": "Godfather 2",
    "runner": "wine",
    "directory": "/run/media/turpster/Files/Games/Godfather II/drive_c/Program Files (x86)/KNIGHT/The Godfather II"
  },
  {
    "id": 1,
    "slug": "origin",
    "name": "Origin",
    "runner": "wine",
    "directory": "/home/turpster/Games/origin"
  },
  {
    "id": 3,
    "slug": "photoshop-cc-2020",
    "name": "Photoshop CC",
    "runner": "wine",
    "directory": ""
  },
  {
    "id": 4,
    "slug": "wolfenstein",
    "name": "Wolfenstein",
    "runner": "wine",
    "directory": "/home/turpster/Games/Wolfenstein/drive_c/Program Files (x86)/DODI-Repacks/Wolfenstein"
  }
]
2020-08-31 11:06:15,883: Shutting down Lutris
2020-08-31 11:06:15,885: MAME XML written
```
The first line of the output `Unable to load locale dir, translations won't work.` is 52 characters long (including the linebreak) and tells me that there is no logic to parse this data and so the add on does not load.

**Comment by Turpster** [comment]
Whilst outputting `result` in the function [get_games()](https://github.com/RobLoach/lutris-kodi-addon/blob/master/default.py), I get this.

```
...
2020-08-31 15:03:33.758 T:140371028727360 NOTICE: [
                                              {
                                                "id": 2,
                                                "slug": "bulletstorm-full-clip-edition",
                                                "name": "Bulletstorm Full Clip Edition",
                                                "runner": "winesteam",
                                                "directory": "/run/media/turpster/Files/Games/steamapps/common/Bulletstorm Full Clip Edition"
                                              },
                                              {
                                                "id": 5,
                                                "slug": "godfather-2",
                                                "name": "Godfather 2",
                                                "runner": "wine",
                                                "directory": "/run/media/turpster/Files/Games/Godfather II/drive_c/Program Files (x86)/KNIGHT/The Godfather II"
                                              },
                                              {
                                                "id": 1,
                                                "slug": "origin",
                                                "name": "Origin",
                                                "runner": "wine",
                                                "directory": "/home/turpster/Games/origin"
                                              },
                                              {
                                                "id": 3,
                                                "slug": "photoshop-cc-2020",
                                                "name": "Photoshop CC",
                                                "runner": "wine",
                                                "directory": ""
                                              },
                                              {
                                                "id": 4,
                                                "slug": "wolfenstein",
                                                "name": "Wolfenstein",
                                                "runner": "wine",
                                                "directory": "/home/turpster/Games/Wolfenstein/drive_c/Program Files (x86)/DODI-Repacks/Wolfenstein"
                                              }
                                            ]
                                            Unable to load locale dir, translations won't work.
...
```
Seems like the message is ending up at the end of the result somehow 🤔

**Comment by Turpster** [comment]
I made a PR to trim `result` so that it only contains JSON data.

**Comment by solbero** [comment]
There seems to be three separate issues going on here.
1. Lutris is notifying you that it is unable to load the locale directory. What locale are you using on your system? Is the English locale installed? I am not able to reproduce this error on my system, probably due to Lutris finding the expected locale.
2. Lutris is returning an error string in the JSON output. It should't do that. This should be reported as an issue to Lutris.
3. The add-on should validate and filter the output from before converting it to a Python array.

**Comment by Turpster** [comment]
Results were the same after locale-gen and restart. 


![image](https://user-images.githubusercontent.com/25931565/91776660-bd0ceb00-ebe5-11ea-8b02-a7ca50ed0f69.png)

**Comment by Turpster** [comment]
Rough guess could be that we are using different operating systems / desktop environments for the warning to not show up for you.

**Comment by solbero** [comment]
Could you log this issue with Lutris not outputting valid JSON with the Lutris client repo and reference this issue there, @Turpster?

**Comment by solbero** [comment]
You fix has been added to the latest version of the add-on.

**Comment by solbero** [comment]
The problem with Lutris outputting an error message in the JSON output is fixed upstream. Here is the commit that fixes it: lutris/lutris@fe01f93

**Comment by solbero** [comment]
I am reopening this issue since the merged fix introduces errors if any of the values in the Lutris JSON contains square brackets. See Issue https://github.com/RobLoach/lutris-kodi-addon/issues/46#issuecomment-718899456 for more information.

The problem you are having with Lutris not outputting valid JSON is fixed in Lutris master with Commit https://github.com/lutris/lutris/commit/fe01f93f85371e105fa288187e974294f2fab377. I'll leave the fix for now in until the next Lutris version is released.
